R/simulation-batch.R

```diff
@@ -19,36 +20,74 @@ SimulationBatch <- R6::R6Class(
     initialize = function(ref, simulation) {
       validateIsOfType(simulation, Simulation)
       super$initialize(ref)
-      self$simulation <- simulation
+      private$.simulation <- simulation
     },
 
-    #' @description Set the parameter and initial values in the simulation and run the simulation
+    #' @description Add a set of parameter and start values for next execution.
+    #' @details Intended for the use with \code{runSimulationBatches}. The simulation batch is executed
+    #' with the sets of parameter and initial values that have been scheduled. The set of run values is cleared after successful run.
+    #'
     #' @param parameterValues Vector of parameter values to set in the simulation (default is `NULL`)
     #' @param initialValues Vector of initial values to set in the simulation  (default is `NULL`)
-    #' @return A `SimulationResults` object containing the result of the simulation run
-    run = function(parameterValues = NULL, initialValues = NULL) {
+    #'
+    #' @return Id of the values set that can be used to get the correct result from \code{runSimulationBatches}.
+    #' @export
+    #'
+    #' @examples
+    #' \dontrun{
+    #' sim <- loadSimulation("simple", loadFromCache = TRUE)
+    #' parameters <- c("Organism|Liver|Volume", "R1|k1")
+    #' molecules <- "Organism|Liver|A"
+    #' # Create two simulation batches. In this case, they are based on the same simulation
+    #' # and vary the same paramter and initial values.
+    #' simulationBatch1 <- createSimulationBatch(simulation = sim,
+    #' parametersOrPaths = parameters,
+    #' moleculesOrPaths = molecules)
+    #' simulationBatch2 <- createSimulationBatch(simulation = sim,
+    #' parametersOrPaths = parameters,
+    #' moleculesOrPaths = molecules)
+    #' #Ids of run values
+    #' ids <- c()
+    #' ids[[1]] <- simulationBatch1$addRunValues(parameterValues = c(1, 2), initialValues = 1)
+    #' ids[[2]] <- simulationBatch1$addRunValues(parameterValues = c(1.6, 2.4), initialValues = 3)
+    #' ids[[3]] <- simulationBatch2$addRunValues(parameterValues = c(4, 2), initialValues = 4)
+    #' ids[[4]] <- simulationBatch2$addRunValues(parameterValues = c(2.6, 4.4), initialValues = 5)

```

Comment:
unclear from this example why you would need TWO simulation batches and not only one with two differents sets.

What is the difference?

Reply:
Was just an example with multiple batches. But as it is `dontrun` anyways, I update the example with two different simulations.